mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/repositories/AvtaleRepository.kt

```diff
@@ -178,18 +181,26 @@ class AvtaleRepository(private val db: Database) {
                    nav_enhet.navn as nav_enhet_navn,
                    t.navn as tiltakstype_navn,
                    t.tiltakskode,
-                   array_agg(distinct lva.organisasjonsnummer) as leverandorUnderenheter,
-                   array_agg(e.enhetsnummer) as navEnheter,
+                   au.leverandor_underenheter,
+                   an.nav_enheter,
                    aa.navident
             from avtale a
                      join tiltakstype t on t.id = a.tiltakstype_id
                      left join avtale_ansvarlig aa on a.id = aa.avtale_id
                      left join nav_enhet on a.nav_region = nav_enhet.enhetsnummer
-                     left join avtale_nav_enhet e on e.avtale_id = a.id
-                     left join avtale_underleverandor lva on lva.avtale_id = a.id
+                     left join lateral (
+                         SELECT an.avtale_id, jsonb_strip_nulls(jsonb_agg(jsonb_build_object('enhetsnummer', an.enhetsnummer, 'navn', ne.navn))) as nav_enheter
+                         FROM avtale_nav_enhet an left join nav_enhet ne on ne.enhetsnummer = an.enhetsnummer WHERE an.avtale_id = a.id GROUP BY 1
+                     ) an on true
+                     left join lateral (
+                         SELECT au.avtale_id, jsonb_strip_nulls(jsonb_agg(jsonb_build_object('organisasjonsnummer', au.organisasjonsnummer, 'navn', v.navn))) as leverandor_underenheter
+                         FROM avtale_underleverandor au left join virksomhet v on v.organisasjonsnummer = au.organisasjonsnummer WHERE au.avtale_id = a.id GROUP BY 1
+                     ) au on true
```

Comment:
På et tidspunkt vil det kanskje lønne seg å splitte opp spørringen her fordi sql'en blir for kompleks... :shrug: